src/System.Linq/src/System/Linq/OrderedEnumerable.cs

```diff
@@ -71,6 +71,57 @@ public List<TElement> ToList()
             return list;
         }
 
+        public HashSet<TElement> ToHashSet(IEqualityComparer<TElement> comparer)
+        {
+            Buffer<TElement> buffer = new Buffer<TElement>(_source);
```

Comment:
Yes, considering that there's no promise with `source.ToHashSet()` that it would always have the exact same results as `new HashSet<T>(source)` that obscure fact about order being retained for immediate enumeration isn't to be expected, and there hasn't been much time for people to hang a dependency on it. This should definitely skip the ordering entirely.